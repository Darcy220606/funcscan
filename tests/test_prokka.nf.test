nextflow_pipeline {

    name "Test pipeline: NFCORE_FUNCSCAN"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_funcscan"
    tag "test_prokka"

    test("test_prokka_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },

                // AMPir
                { assert snapshot("$outputDir/amp/ampir/sample_1_short/sample_1_short.ampir.tsv").match("ampir_tsv_1-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_1_short/sample_1_short.ampir.faa").match("ampir_faa_1-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_short/sample_2_short.ampir.tsv").match("ampir_tsv_2-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_short/sample_2_short.ampir.faa").match("ampir_faa_2-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_long/sample_2_long.ampir.tsv").match("ampir_tsv_2-long") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_long/sample_2_long.ampir.faa").match("ampir_faa_2-long") },

                // AMPlify
                { assert snapshot("$outputDir/amp/amplify/sample_1_short/sample_1_short.tsv").match("amplify_tsv_1-short") },
                { assert snapshot("$outputDir/amp/amplify/sample_2_short/sample_2_short.tsv").match("amplify_tsv_2-short") },
                { assert snapshot("$outputDir/amp/amplify/sample_2_long/sample_2_long.tsv").match("amplify_tsv_2-long") },

                // HMMsearch
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_1_short/sample_1_short_mybacteriocin.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_2_short/sample_2_short_mybacteriocin.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_2_long/sample_2_long_mybacteriocin.txt.gz").exists() },

                // Macrel
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short.smorfs.faa.gz").match("macrel_smorfs_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short.smorfs.faa.gz").match("macrel_smorfs_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long.smorfs.faa.gz").match("macrel_smorfs_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short.all_orfs.faa.gz").match("macrel_all_orfs_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short.all_orfs.faa.gz").match("macrel_all_orfs_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long.all_orfs.faa.gz").match("macrel_all_orfs_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short.prediction.gz").match("macrel_amp_prediction_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short.prediction.gz").match("macrel_amp_prediction_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long.prediction.gz").match("macrel_amp_prediction_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/README.md").match("macrel_readme_file_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/README.md").match("macrel_readme_file_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/README.md").match("macrel_readme_file_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short_log.txt").match("macrel_log_file_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short_log.txt").match("macrel_log_file_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long_log.txt").match("macrel_log_file_2-long") },

                // ampcombi
                { assert new File("$outputDir/reports/ampcombi/ampcombi_complete_summary.tsv").exists() }, // check again with new ampcombi version, if md5 check is possible

                // DeepARG
                { assert snapshot("$outputDir/arg/deeparg/sample_1_short/sample_1_short.align.daa.tsv").match("deeparg_daa_tsv_1-short") },
                { assert snapshot("$outputDir/arg/deeparg/sample_2_short/sample_2_short.align.daa.tsv").match("deeparg_daa_tsv_2-short") },
                { assert snapshot("$outputDir/arg/deeparg/sample_2_long/sample_2_long.align.daa.tsv").match("deeparg_daa_tsv_2-long") },
                { assert snapshot("$outputDir/arg/deeparg/sample_1_short/sample_1_short.mapping.ARG").match("deeparg_daa_arg_1-short") },
                { assert snapshot("$outputDir/arg/deeparg/sample_2_short/sample_2_short.mapping.ARG").match("deeparg_daa_arg_2-short") },
                { assert snapshot("$outputDir/arg/deeparg/sample_2_long/sample_2_long.mapping.ARG").match("deeparg_daa_sarg_2-long") },
                { assert file("$outputDir/arg/deeparg/sample_1_short/sample_1_short.align.daa").name },
                { assert file("$outputDir/arg/deeparg/sample_2_short/sample_2_short.align.daa").name },
                { assert file("$outputDir/arg/deeparg/sample_2_long/sample_2_long.align.daa").name },
                { assert path("$outputDir/arg/deeparg/sample_1_short/sample_1_short.mapping.potential.ARG").text.contains("#ARG") },
                { assert path("$outputDir/arg/deeparg/sample_2_short/sample_2_short.mapping.potential.ARG").text.contains("#ARG") },
                { assert path("$outputDir/arg/deeparg/sample_2_long/sample_2_long.mapping.potential.ARG").text.contains("#ARG") },

                // ABRicate
                { assert snapshot("$outputDir/arg/deeparg/sample_1_short/sample_1_short.txt").match("abricate_txt_1-short") },
                { assert snapshot("$outputDir/arg/deeparg/sample_2_short/sample_2_short.txt").match("abricate_txt_2-short") },
                { assert snapshot("$outputDir/arg/deeparg/sample_2_long/sample_2_long.txt").match("abricate_txt_2-long") },

                // AMRFinderPlus
                { assert snapshot("$outputDir/arg/amrfinderplus/sample_1_short/sample_1_short.tsv").match("amrfinderplus_tsv_1-short") },
                { assert snapshot("$outputDir/arg/amrfinderplus/sample_2_short/sample_2_short.tsv").match("amrfinderplus_tsv_2-short") },
                { assert snapshot("$outputDir/arg/amrfinderplus/sample_2_long/sample_2_long.tsv").match("amrfinderplus_tsv_2-long") },

                // RGI
                { assert snapshot("$outputDir/arg/rgi/sample_1_short/sample_1_short.txt").match("rgi_txt_1-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_short/sample_2_short.txt").match("rgi_txt_2-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_long/sample_2_long.txt").match("rgi_txt_2-long") },

                // fARGene
                { assert snapshot("$outputDir/arg/rgi/sample_1_short/class_a/results_summary.txt").match("fargene_class_a_txt_1-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_short/class_a/results_summary.txt").match("fargene_class_a_txt_2-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_long/class_a/results_summary.txt").match("fargene_class_a_txt_2-long") },
                { assert snapshot("$outputDir/arg/rgi/sample_1_short/class_b_1_2/results_summary.txt").match("fargene_class_b_txt_1-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_short/class_b_1_2/results_summary.txt").match("fargene_class_b_txt_2-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_long/class_b_1_2/results_summary.txt").match("fargene_class_b_txt_2-long") },
                { assert snapshot("$outputDir/arg/rgi/sample_1_short/class_b_1_2/fargene_analysis.log").match("fargene_log_1-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_short/class_b_1_2/fargene_analysis.log").match("fargene_log_2-short") },
                { assert snapshot("$outputDir/arg/rgi/sample_2_long/class_b_1_2/fargene_analysis.log").match("fargene_log_2-long") },

                // hamronization
                { assert snapshot("$outputDir/reports/hamronization_summarize/hamronization_combined_report.tsv").match("hamronization_summarize") },
            )
        }
    }
}
