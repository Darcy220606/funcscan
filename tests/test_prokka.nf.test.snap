{
    "hamronization_summarize": {
        "content": [
            "hamronization_combined_report.tsv:md5,864466b0fb1acfc0e6b3425271f78ecb"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.634478107"
    },
    "abricate": {
        "content": [
            "sample_1.txt:md5,69af3321b0bc808b7ef85f102395736f",
            "sample_2.txt:md5,69af3321b0bc808b7ef85f102395736f"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.565623351"
    },
    "fargene": {
        "content": [
            "results_summary.txt:md5,690d351cfc52577263ef4cfab1c81f50",
            "results_summary.txt:md5,690d351cfc52577263ef4cfab1c81f50",
            "results_summary.txt:md5,f69af9b0e18a0cfc934eb18b7e4bffe1",
            "results_summary.txt:md5,f69af9b0e18a0cfc934eb18b7e4bffe1"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.619183937"
    },
    "rgi": {
        "content": [
            "sample_1.txt:md5,fce130af51f93cccfc09ddaf9caf623f",
            "sample_2.txt:md5,fce130af51f93cccfc09ddaf9caf623f"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.604862932"
    },
    "deeparg_tsv_ARG": {
        "content": [
            "sample_1.align.daa.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_2.align.daa.tsv:md5,06648de08caca0b7f42eab9576615226",
            "sample_1.mapping.ARG:md5,0e049e99eab4c55666062df21707d5b9",
            "sample_2.mapping.ARG:md5,0e049e99eab4c55666062df21707d5b9"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.543230941"
    },
    "ampir": {
        "content": [
            "sample_1.ampir.tsv:md5,77e9ed892f8a2783248ac015a30671fd",
            "sample_1.ampir.faa:md5,af06d0d5dac80983c40d045f23b222e2",
            "sample_2.ampir.tsv:md5,d426c2662bde91dd581377aaef189d38",
            "sample_2.ampir.faa:md5,53e3bd83815fdab4974d943e8fe64b9c"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.460611446"
    },
    "ampcombi_logfiles": {
        "content": [
            "Ampcombi_cluster.log:md5,4c78f5f134edf566f39e04e3ab7d8558",
            "Ampcombi_complete.log:md5,3dabfea4303bf94bd4f5d78c5b8c83c1",
            "Ampcombi_parse_tables.log:md5,1e2b5abad7d17e03428066f345b91117"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.5216115"
    },
    "amplify": {
        "content": [
            "sample_1.amplify.tsv:md5,cc87ab238978f1ab20179bd63057323e",
            "sample_2.amplify.tsv:md5,a452acb268aaa97f08c7896567c143db"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.47570837"
    },
    "macrel": {
        "content": [
            "sample_1.macrel.smorfs.faa.gz:md5,1b5e2434860e635e95324d1804a3be7b",
            "sample_2.macrel.smorfs.faa.gz:md5,38108b5cdfdc2196afe67418b9b04682",
            "sample_1.macrel.all_orfs.faa.gz:md5,86f6b3b590d1b22d9c5aa164f8a14080",
            "sample_2.macrel.all_orfs.faa.gz:md5,fdb384925af50ecade05dccaff68afd8",
            "sample_1.macrel.prediction.gz:md5,0c4b16e0838be56e012b99169863a168",
            "sample_2.macrel.prediction.gz:md5,440deffd6b6d9986ce098e44c66db9ae",
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb",
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb",
            "sample_1.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_2.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.486210507"
    },
    "amrfinderplus": {
        "content": [
            "sample_1.tsv:md5,b4d261ace9be7d013c19d1f5c0005bfe",
            "sample_2.tsv:md5,b4d261ace9be7d013c19d1f5c0005bfe"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-25T18:46:33.583863983"
    }
}